libs/ngx-dsfr/search-bar/src/lib/search-bar.component.ts

```diff
@@ -58,8 +58,15 @@ export interface ItemResult {
   ]
 } )
 export class DsfrSearchBarComponent implements ControlValueAccessor, OnDestroy, OnChanges, AfterViewInit {
+  searchInput!: ElementRef<HTMLInputElement>;
 
-  @ViewChild( 'searchInput' ) searchInput!: ElementRef<HTMLInputElement>;
+  @ViewChild( 'inputWithoutSearchButton', { static: false } ) set inputWithoutSearchButtonContent( content: ElementRef<HTMLInputElement> ) {
+    this.setSearchInput( content );
+  }
+
+  @ViewChild( 'inputWithSearchButton', { static: false } ) set inputWithSearchButtonContent( content: ElementRef<HTMLInputElement> ) {
+    this.setSearchInput( content );
+  }
```

Comment:
What was wrong with `#searchInput` for both occurrences of the `<input>` in the Template ?

They can't exist simultaneously and they don't seem to be treated differently in what you're doing here...

So rather than making the code more complicated with these 2 setters, can you test if only one works ?